src/streamlink_cli/main.py

```diff
@@ -56,12 +56,16 @@ def check_file_output(filename, force):
 
     log.debug("Checking file output")
 
-    if os.path.isfile(filename) and not force:
-        answer = console.ask("File {0} already exists! Overwrite it? [y/N] ",
-                             filename)
-
-        if answer.lower() != "y":
-            sys.exit()
+    if sys.stdin.isatty():
+        if os.path.isfile(filename) and not force:
+            answer = console.ask("File {0} already exists! Overwrite it? [y/N] ",
+                                 filename)
+
+            if answer.lower() != "y":
+                sys.exit()
+    elif os.path.isfile(filename):
```

Comment:
`--force` won't work, ` and not force` is required

instead of **repeating** it, it could be called first and isatty could be used after it

```py
    if os.path.isfile(filename) and not force:
        if sys.stdin.isatty():
            answer = console.ask("File {0} already exists! Overwrite it? [y/N] ",
                                 filename)

            if answer.lower() != "y":
                sys.exit()
        else:
            log.error("File {0} already exists, use --force to overwrite it.".format(filename))
            sys.exit()
```

Reply:
Yep. My bad :)